### ISSUE TITLE
duffle build should add a digest to invocation image

### ISSUE THREAD

**Issue by michelleN** [deislabs/duffle#690]
Docker image support only for now. Once we have digest implementation for all parts of the duffle lifecycle, we can then go on to [support VM images](#689 )

**Comment by jeremyrickard** [comment]
I think that we should _instead_ implement this as part of _duffle push_

**Comment by jeremyrickard** [comment]
Per a conversation with @radu-matei in #cnab 

1.) When we `duffle push` a bundle that hasn't been pushed, we should first:

**Comment by glyn** [comment]
I don't think it is necessary to generate a relocation mapping file when we `duffle push` a bundle to another registry. According to a [cnab-to-oci issue](https://github.com/docker/cnab-to-oci/issues/58#issuecomment-519066417), the relocation mapping file can be generated as needed from the registry.

(Also, I'm a little confused because I don't see `duffle push` at master. Is there a pre-req to this issue? Presumably `duffle push` isn't a 1.0 feature because it depends on the stilll-to-be-written CNAB registries spec.)

**Comment by jeremyrickard** [comment]
@glyn yes, `duffle push` is not in master. @radu-matei has been working on _a version_ of it in another PR #681 and I wanted to capture this. The original issue was quite old, so I wanted to update it with discussion from slack.

**Comment by glyn** [comment]
We should also keep in mind cases where the bundle is _never_ pushed, for example:
1. Build a bundle
2. `docker push` the invocation image
3. Export the bundle as a thick bundle

Exporting the bundle will effectively "freeze" the invocation image and therefore its content digest. But the bundle may still have an undigested reference to the invocation image and could therefore end up executing a different image.

**Comment by radu-matei** [comment]
Circling back on the point of generating a relocation map as part of the `push` process I think I agree with @glyn - we can generate the relocation map based on the information in the OCI registry - stored `bundle.json` and the OCI index.

#828 deals with the workflow `build` -> `export`.
